python/GafferDispatchUI/DispatchDialogue.py

```diff
@@ -58,15 +58,15 @@ class DispatchDialogue( GafferUI.Dialogue ) :
 
 	__dispatchDialogueMenuDefinition = None
 
-	def __init__( self, tasks, dispatchers, nodes, postDispatchBehaviour=PostDispatchBehaviour.Confirm, title="Dispatch Tasks", sizeMode=GafferUI.Window.SizeMode.Manual, **kw ) :
+	def __init__( self, tasks, dispatchers, nodes=None, postDispatchBehaviour=PostDispatchBehaviour.Confirm, title="Dispatch Tasks", sizeMode=GafferUI.Window.SizeMode.Manual, **kw ) :
```

Comment:
This doesn't provide full backwards compatibility, because we've still added a new positional argument before `postDispatchBehaviour`. For full compatibility, the nodes argument would need to be added after all the others, which is a bit of a pity, because it's definitely a more important argument than `postDispatchBehaviour`. If you're not in desperate need of a backport, I'd favour the clean break.

As an aside, it seems that Python 3 has [keyword-only arguments](https://www.python.org/dev/peps/pep-3102/), which could help with this sort of problem. I could see myself using them not just to help with this type of problem, but also to force call sites to be more readable - it's often easier to understand keyword arguments than positional ones.

Reply:
OK, but we can't backport without it being fully API-compatible. For master I think what we have already is ideal, so unless a backport is absolutely essential, I'd go with that..

Reply:
Hmm, just coming out of my vacation brain, perhaps this is non-backportable but still an improvement. In the complex case that motivated the PR, we want to specify different tasks vs UI nodes. But in the simpler case of showing the node you’re dispatching, you’d need to double-specify it without this (and the previous) commit. Thoughts?

Reply:
I thought this was an improvement too, but thinking more about it I don't think it's quite right. There should definitely be a way of preventing any nodes from being shown, but currently there isn't, either in the DispatchDialogue or the app.

For the DispatchDialogue we could have `nodes = None` be the default, and mean "show the tasks", with the ability to pass `nodes = []` to say "show nothing". For the app it's harder, since we don't have any ability to distinguish between "no list" and an empty list. Maybe a special default of "<tasks>" could mean show the tasks, and then an empty list can be used for "show nothing"?

Reply:
Alternatively, I think I'm actually fine with having to pass the same list to each argument if I want to show all the tasks - at least it's nice and explicit. If we did that though, we'd need to allow a `-nodes` argument even when there's no script in the dispatch app, so one can do `-tasks GafferDispatch.SystemCommand -nodes SystemCommand`.

Reply:
Addressed in 4aa3d189a67066e9dcf436b01c1ca4ec0ac53b90